src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -552,9 +576,119 @@ private static ITaskSeriesTimer StartParameterLogTimer(IRecurrentCommand updateC
                 await TryHandleTimeoutAsync(invokeTask, CancellationToken.None, throwOnTimeout, timeoutTokenSource.Token, timerInterval, instance, null);
             }
 
-            object returnValue = await invokeTask;
+            if (invokeTask != null)
```

Comment:
The test Microsoft.Azure.WebJobs.Host.EndToEndTests.AsyncChainEndToEndTests.TraceWriterLogging doesn't log properly and the test fails when the check for null is removed. This is something we can further discuss.